scripts/go-function-build.sh

```diff
@@ -2,9 +2,26 @@
 
 set -euo pipefail
 
-image_tag=${TAG:-dev}
-prefix="release-go-functions-"
-[[ "${image_tag}" = "${prefix}"* ]] && image_tag="${image_tag#$prefix}"
-image_name=gcr.io/kpt-functions/"${CURRENT_FUNCTION}"
-image="${image_name}":"${image_tag}"
-docker build -t "${image}" -t "${image_name}" -f "${CURRENT_FUNCTION}"/Dockerfile "${CURRENT_FUNCTION}"
+scripts_dir="$(dirname "$0")"
+# git-tag-parser.sh has been shell-checked separately.
+# shellcheck source=/dev/null
+source ${scripts_dir}/git-tag-parser.sh
+
+DEV_TAG=dev
+GIT_TAG=${GIT_TAG:-"${TAG}"}
+versions=$(get_versions "${GIT_TAG}")
+fn_name=$(get_fn_name "${GIT_TAG}")
+fn_name=${fn_name:-"${CURRENT_FUNCTION}"}
+GCR_REGISTRY=${GCR_REGISTRY:-gcr.io/kpt-functions}
+
+cd "${scripts_dir}"/../functions/go
+docker build -t "${GCR_REGISTRY}/${fn_name}:${DEV_TAG}" -f "${fn_name}"/Dockerfile "${fn_name}"
+
+# TODO(https://github.com/GoogleContainerTools/kpt/issues/1368): stop building
+# docker images with tag "latest".
+echo tagging "${GCR_REGISTRY}/${fn_name}:latest"
+docker tag "${GCR_REGISTRY}/${fn_name}:${DEV_TAG}" "${GCR_REGISTRY}/${fn_name}:latest"
```

Comment:
Any special reason to hardcode the latest tag here? Can't we add a step in the `.github/ci.yaml` to build latest tag images?

Reply:
>Any special reason to hardcode the latest tag here? 

No special reason. But I prefer to be explicit.

> Can't we add a step in the .github/ci.yaml to build latest tag images?

We have a step `make build` in `ci.yaml`. `make build` invokes `go-function-build.sh` underlying.
Why suggesting add the step in `ci.yaml`?

Reply:
Add a `make TAG=latest build` so that we don't need to add a un-skippable latest tag here.

Reply:
I assume the build cache is shared across steps in GitHub Actions. It should not double the build time.

Reply:
It's not about the build time. When I build images locally I may only want to build dev tag, or only latest tag. This can be controlled by make command. If latest is hardcoded in the script, I cannot avoid building it if I don't want it.